Service/src/main/java/org/apidb/apicommon/service/services/jbrowse/JBrowseUserDatasetsService.java

```diff
@@ -1,161 +1,206 @@
+import org.apidb.apicommon.service.services.jbrowse.model.JBrowseTrack;
+import org.apidb.apicommon.service.services.jbrowse.model.VDIDatasetReference;
+import org.apidb.apicommon.service.services.jbrowse.model.VDIDatasetType;
+import org.gusdb.fgputil.db.runner.SQLRunner;
 import org.gusdb.fgputil.events.Events;
 import org.gusdb.wdk.errors.ErrorContext.ErrorLocation;
 import org.gusdb.wdk.errors.ServerErrorBundle;
 import org.gusdb.wdk.events.ErrorEvent;
 import org.gusdb.wdk.model.WdkModelException;
 import javax.ws.rs.ForbiddenException;
-import org.gusdb.wdk.model.user.dataset.UserDataset;
-import org.gusdb.wdk.model.user.dataset.UserDatasetDependency;
-import org.gusdb.wdk.model.user.dataset.UserDatasetFile;
-import org.gusdb.wdk.model.user.dataset.UserDatasetInfo;
-import org.gusdb.wdk.model.user.dataset.UserDatasetMeta;
-import org.gusdb.wdk.model.user.dataset.UserDatasetSession;
-import org.gusdb.wdk.model.user.dataset.UserDatasetType;
-import org.gusdb.wdk.service.formatter.UserDatasetsFormatter;
-import org.gusdb.wdk.service.service.user.UserDatasetService;
+import javax.ws.rs.core.Response;
+
 import org.gusdb.wdk.service.service.user.UserService;
-import org.json.JSONArray;
-import org.json.JSONObject;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.gusdb.wdk.service.FileRanges.getFileChunkResponse;
+import static org.gusdb.wdk.service.FileRanges.parseRangeHeaderValue;
 
 public class JBrowseUserDatasetsService extends UserService {
 
   private static final Logger LOG = Logger.getLogger(JBrowseUserDatasetsService.class);
+  private static final String VDI_DATASET_DIR_KEY = "VDI_DATASETS_DIRECTORY";
+  private static final String VDI_CONTROL_SCHEMA_KEY ="VDI_CONTROL_SCHEMA";
+  public static final String BIGWIG_SUBCATEGORY = "Bigwig Files From User";
+  public static final String RNA_SEQ_SUBCATEOGRY = "RNASeq";
 
   public JBrowseUserDatasetsService(@PathParam(USER_ID_PATH_PARAM) String uid) {
     super(uid);
   }
 
+
+  /**
+   * This endpoint exposes user dataset track files with the intent of JBrowse using it to retrieve file ranges.
+   * Any installed file for a dataset owned by the user can be fetched by this endpoint.
+   *
+   * URLs to hit this endpoint are constructed by this service when the JBrowse application asks for the user's available
+   * tracks.
+   */
   @GET
-  @Path("user-datasets-jbrowse/{organism}")
+  @Path("user-datasets-jbrowse/data")
   @Produces(MediaType.APPLICATION_JSON)
-  public JSONObject getAllUserDatasetsJBrowse(@PathParam("organism") String publicOrganismAbbrev) {
+  public Response getAllUserDatasetFileJBrowse(@QueryParam("data") String data,
```

Comment:
If the "data" query param is supposed to be the UD ID then maybe a different name? Or looks like it may be a path that could have separate segments? Maybe different name then too.